### ISSUE TITLE
Comment that should be kept and its subsequent rule are both removed when following a rule with a double hyphen selector

### ISSUE THREAD

**Issue by jovilog** [clean-css/clean-css#1230]
I believe this fix 27103dece874460e58f14693872e411482c491b8 breaks the following CSS, because this works as expected with `5.2.2`. Probably related to #1218, #1224.

### Environment

* clean-css version - `>5.2.3`:
* node.js version - `16.13.0`:
* operating system: Windows 10

### CL output

```
WARNING: Invalid selector '/*!
 * Please, keep this comment.
 */
.disappears' at test.css:2:0. Ignoring.
```

### Of importance

The problem does NOT occur if:

- the selector has no double hyphen
- the rule with the double hyphen ends with a `;` before the closing `}`
- there's another rule before the comment that ends with a `;` before the closing `}`
- the comment is not of the sort to keep, i.e. without the `!`

### Input CSS

```css
.double--hyphen{color:blue}
/*!
 * Please, keep this comment.
 */
.disappears{color:red}
.stays{color:green}
```

### Actual output CSS

```css
.double--hyphen{color:#00f}.stays{color:green}
```

### Expected output CSS

```css
.double--hyphen{color:#00f}/*!
 * Please, keep this comment.
 */.disappears{color:red}.stays{color:green}
```

**Comment by jakubpawlowicz** [comment]
Fixed alongside #1224.

**Comment by jakubpawlowicz** [comment]
The fix is out in 5.3.2.
